pkg/vm/vm.go

```diff
@@ -1214,15 +1214,15 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 			panic("invalid length")
 		}
 
-		items := make([]stackitem.MapElement, n)
-		for i := range n {
-			key := v.estack.Pop()
-			validateMapKey(key)
+		m := stackitem.NewMap()
+		for range n {
+			key := v.estack.Pop().value
 			val := v.estack.Pop().value
-			items[i].Key = key.value
-			items[i].Value = val
+			if key != nil {
+				m.Add(key, val)
+			}
```

Comment:
The behaviour must be compatible with the old implementation. Review the implementation of `validateMapKey`.